src/Engine/Physics/TriggerObserver.cpp

```diff
@@ -90,14 +90,26 @@ namespace Physics {
         enterHandler = handler;
     }
 
+    void TriggerObserver::ForgetEnter() {
+        enterHandler = nullptr;
+    }
+
     void TriggerObserver::OnRetain(const std::function<void()>& handler) {
         retainHandler = handler;
     }
 
+    void TriggerObserver::ForgetRetain() {
+        retainHandler = nullptr;
+    }
+
     void TriggerObserver::OnLeave(const std::function<void()>& handler) {
         leaveHandler = handler;
     }
 
+    void TriggerObserver::ForgetLeave() {
+        leaveHandler = nullptr;
+    }
+
     // Called with each contact for our own processing. This is where we can
     // provide custom intersection handling.
     btScalar TriggerObserver::addSingleResult(btManifoldPoint& cp,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'addSingleResult' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)